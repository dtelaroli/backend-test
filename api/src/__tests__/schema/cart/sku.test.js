const { seed, mutate } = require("../../helpers");
const { SKU, PRODUCT } = require("../../../db/seeders/consts");

const {
  mutations: { CREATE_SKU, UPDATE_SKU, DELETE_SKU },
} = require("../graphql");

describe("Sku", () => {
  beforeEach(() => {
    seed();
  });

  describe("Mutation", () => {
    const ID = "11111111-1111-1111-1111-111111111110";

    describe("createProduct", () => {
      it("Should create and delete sku", async () => {
        const create = await mutate({
          mutation: CREATE_SKU,
          variables: {
            input: {
              id: ID,
              sku: "S123",
              inventory: 10,
              price: 9.99,
              productId: PRODUCT.id,
            },
          },
        });

        const destroy = await mutate({
          mutation: DELETE_SKU,
          variables: {
            id: ID,
          },
        });

        expect({ create, destroy }).toMatchSnapshot();
      });

      it("Should update sku", async () => {
        const result = await mutate({
          mutation: UPDATE_SKU,
          variables: {
            input: {
              id: SKU.id,
              sku: "S123",
              inventory: 5,
              price: 4.12,
            },
          },
        });

        expect(result).toMatchSnapshot();
      });
    });
  });
});
