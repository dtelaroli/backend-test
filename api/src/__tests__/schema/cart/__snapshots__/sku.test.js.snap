// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sku Mutation createProduct Should create and delete sku 1`] = `
Object {
  "create": Object {
    "data": Object {
      "createSku": Object {
        "id": "11111111-1111-1111-1111-111111111110",
        "inventory": 10,
        "price": 9.99,
        "product": Object {
          "id": "11111111-1111-1111-1111-111111111113",
          "image": "image1.jpg",
          "name": "product1",
        },
        "sku": "S123",
      },
    },
    "errors": undefined,
    "extensions": undefined,
    "http": Object {
      "headers": Headers {
        Symbol(map): Object {
          "Cache-Control": Array [
            "max-age=5, public",
          ],
        },
      },
    },
  },
  "destroy": Object {
    "data": Object {
      "deleteSku": Object {
        "message": "Sku removed",
      },
    },
    "errors": undefined,
    "extensions": undefined,
    "http": Object {
      "headers": Headers {
        Symbol(map): Object {
          "Cache-Control": Array [
            "max-age=5, public",
          ],
        },
      },
    },
  },
}
`;

exports[`Sku Mutation createProduct Should update sku 1`] = `
Object {
  "data": Object {
    "updateSku": Object {
      "id": "11111111-1111-1111-1111-111111111112",
      "inventory": 5,
      "price": 4.12,
      "product": Object {
        "id": "11111111-1111-1111-1111-111111111113",
        "image": "image1.jpg",
        "name": "product3",
      },
      "sku": "S123",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {
        "Cache-Control": Array [
          "max-age=5, public",
        ],
      },
    },
  },
}
`;
